api/v1alpha1/release_types.go

```diff
@@ -138,128 +114,128 @@ type Release struct {
 	Status ReleaseStatus `json:"status,omitempty"`
 }
 
-// HasStarted checks whether the Release has a valid start time set in its status.
-func (r *Release) HasStarted() bool {
-	return r.Status.StartTime != nil && !r.Status.StartTime.IsZero()
+// HasDeploymentFinished checks whether the Release has been successfully deployed via GitOps.
+func (r *Release) HasDeploymentFinished() bool {
+	condition := meta.FindStatusCondition(r.Status.Conditions, deployedConditionType.String())
+	return condition != nil && condition.Status != metav1.ConditionUnknown
 }
 
-// HasSucceeded checks whether the Release has succeeded or not.
-func (r *Release) HasSucceeded() bool {
-	return meta.IsStatusConditionTrue(r.Status.Conditions, releaseConditionType)
+// HasReleaseFinished returns a boolean indicating whether the Release's status indicates that it is done or not.
+func (r *Release) HasReleaseFinished() bool {
+	condition := meta.FindStatusCondition(r.Status.Conditions, releasedConditionType.String())
+	return condition != nil && condition.Status != metav1.ConditionUnknown
 }
 
-// IsDeployed checks whether the Release has been successfully deployed via GitOps.
+// IsDeployed checks whether the Deployment has succeeded or not.
 func (r *Release) IsDeployed() bool {
-	condition := meta.FindStatusCondition(r.Status.Conditions, applicationapiv1alpha1.ComponentDeploymentConditionAllComponentsDeployed)
-	return condition != nil && condition.Status == metav1.ConditionTrue
+	return meta.IsStatusConditionTrue(r.Status.Conditions, deployedConditionType.String())
 }
 
 // IsDeploying checks whether the Release has a valid start time for the deployment set in its status.
 func (r *Release) IsDeploying() bool {
-	return r.Status.DeploymentStartTime != nil && !r.Status.DeploymentStartTime.IsZero()
+	return r.Status.Deployment.StartTime != nil && !r.Status.Deployment.StartTime.IsZero()
 }
 
-// IsDone returns a boolean indicating whether the Release's status indicates that it is done or not.
-func (r *Release) IsDone() bool {
-	condition := meta.FindStatusCondition(r.Status.Conditions, releaseConditionType)
-	return condition != nil && condition.Status != metav1.ConditionUnknown
+// IsReleased checks whether the Release has succeeded or not.
```

Comment:
I don't like `isReleased` and `isReleasing` as function names. Telling someone "the release is releasing" is not helpful at all imo

Reply:
They refer to the releasing process. I get your comment though. Propose new names.